diff -uNr --exclude .svn jboss-as.orig/bin/run.conf jboss-as/bin/run.conf
--- jboss-as.orig/bin/run.conf	2010-07-12 11:09:53.000000000 +0200
+++ jboss-as/bin/run.conf	2010-07-12 11:22:05.000000000 +0200
@@ -42,7 +42,7 @@
 # Specify options to pass to the Java VM.
 #
 if [ "x$JAVA_OPTS" = "x" ]; then
-   JAVA_OPTS="-Xms128m -Xmx512m -XX:MaxPermSize=256m -Dorg.jboss.resolver.warning=true -Dsun.rmi.dgc.client.gcInterval=3600000 -Dsun.rmi.dgc.server.gcInterval=3600000"
+   JAVA_OPTS="-Xms512m -Xmx1024m -XX:MaxPermSize=256m -Dorg.jboss.resolver.warning=true -Dsun.rmi.dgc.client.gcInterval=3600000 -Dsun.rmi.dgc.server.gcInterval=3600000"
 fi
 
 # Sample JPDA settings for remote socket debugging
@@ -50,3 +50,6 @@
 
 # Sample JPDA settings for shared memory debugging 
 #JAVA_OPTS="$JAVA_OPTS -Xrunjdwp:transport=dt_shmem,address=jboss,server=y,suspend=n"
+
+# GateIn configuration directory
+JAVA_OPTS="$JAVA_OPTS -Dexo.conf.dir.name=gatein"
